pyproject.toml

```diff
@@ -19,22 +19,19 @@
 # build dependencies should be fixed - including all transitive dependencies. This way we can ensure
 # reproducibility of the build and make sure that any future releases of any dependencies will not
 # break the build of released airflow sources in the future.
-# We should periodically refresh these build dependencies by following this procecedure:
-# 1. Create a new virtual environment
-# 2. Run `pip install hatchling gitpython
-# 3. Run `pip freeze`
-# 4. Copy versions you see here (note that tomli needs to have python_version < '3.11').
+# The dependencies can be automatically upgraded by running:
+# pre-commit run --hook-stage manual update-build-dependencies --all-files
 requires = [
+    "GitPython==3.1.41",
     "editables==0.5",
     "gitdb==4.0.11",
-    "gitpython==3.1.40",
-    "hatchling==1.21.0",
+    "hatchling==1.21.1",
     "packaging==23.2",
     "pathspec==0.12.1",
-    "pluggy==1.3.0",
+    "pluggy==1.4.0",
     "smmap==5.0.1",
-    "tomli==2.0.1; python_version < '3.11'",
-    "trove-classifiers==2023.11.29",
+    "tomli==2.0.1",
```

Comment:
Just wondering should we also add marker `python_version < '3.11'` here?

Reply:
Hmm. that's tricky, but yes I think we should post-process it.

Reply:
Done :)